### PR TITLE
feat(1412): add metrics for job scope

### PR DESCRIPTION
**Pull request by d2lam**
Add metrics for job scope
- Find build duration for a specific job within a time range
- Find step duration across builds for a specific job during within a time range

Related: 
https://github.com/screwdriver-cd/screwdriver/issues/1412

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -534,13 +545,14 @@ class BuildModel extends BaseModel {
      * @param  {Object}   [config]              Configuration object
      * @param  {String}   [config.startTime]    Look at steps created after this startTime
      * @param  {String}   [config.endTime]      Look at steps created before this endTime
-     * @return {Promise}  Resolves to array of metrics for steps belong to this event
+     * @return {Promise}  Resolves to array of metrics for steps belong to this build
```
Comment (tkyi) [inline lib/build.js:556]: need to add JSdoc for `stepName`

Comment (d2lam) [reply lib/build.js:556]: Added

**Review by tkyi — APPROVED** [review]

**Comment by tkyi** [comment]
Not sure if need to update README

**Comment by d2lam** [comment]
@tkyi  Added README
Also added additional id fields in returned metrics, in case the UI needs it for displaying. If it doesn't need, we can come back and remove them.

**Pull request merged**